premailer/cache.py

```diff
@@ -1,83 +1,61 @@
 import functools
 
 
-class _HashedSeq(list):
-    # # From CPython
-    __slots__ = 'hashvalue'
-
-    def __init__(self, tup, hash=hash):
-        self[:] = tup
-        self.hashvalue = hash(tup)
-
-    def __hash__(self):
-        return self.hashvalue
-
-
-# if we only have nonlocal
-class _Cache(object):
-    def __init__(self):
-        self.off = False
-        self.missed = 0
-        self.cache = {}
-
-
-def function_cache(expected_max_entries=1000):
+def function_cache():
```

Comment:
Now is not the time but the correct name ought to be "function_memoize" or something like that.